bionic/datatypes.py

```diff
@@ -72,15 +68,36 @@ class Result(object):
     Represents one value for one entity.
     '''
     def __init__(
-            self, query, value, file_path=None):
+            self, query, value, value_hash=None, file_path=None):
         self.query = query
         self.value = value
+        # Only present when value should persist.
         self.file_path = file_path
+        self.value_hash = value_hash
 
     def __repr__(self):
         return f'Result({self.query!r}, {self.value!r})'
 
 
+class ProvenanceDigest(object):
+    '''
+    A collection of values used by Provenance for different chained hashes.
+    This hashes depend on the entities and can come from either another
```

Comment:
nit:
* s/This hashes/These hashes/ 
* s/or value hash in result/or the value hash of a result/